File: SentiTrack/app/schemas/summary.py
"""Weekly summary request/response schemas."""

from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class WeeklySummaryGenerate(BaseModel):
    week_of: date | None = Field(
        default=None,
        description="Any date within the target week (Mon–Sun). Defaults to the current week.",
        examples=["2026-07-20"],
    )


class WeeklySummaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    week_start: date
    week_end: date
    summary: str
    suggestions: list[str]
    entry_count: int
    created_at: datetime
    updated_at: datetime

File: SentiTrackAiFrontend/src/types/summary.ts
export interface WeeklySummary {
  id: number
  week_start: string
  week_end: string
  summary: string
  suggestions: string[]
  entry_count: number
  created_at: string
  updated_at: string
}

export interface GenerateWeeklySummaryPayload {
  week_of?: string
}
